### PR TITLE
[#9752] Instructor: Course/Student/Edit: Able to exceed section limit by editing section name of an existing student

### LINKED ISSUE #9752: Instructor: Course/Student/Edit: Able to exceed section limit by editing section name of an existing student
- **Environment**: 

`master` branch at commit 4346f6f3d5ef8bf9570ab088a9835709f13ca99f
There's a current limit of 100 per section.

**Steps to reproduce**
1. Log in as instructor
2. Have 1 course with **100 students in Section 1** and **a student in Section 2**. 
3. Go to instructor/courses/student/edit?courseid=ANY_VALID_COURSE_ID&studentemail=EMAIL_OF_SECTION2_STUDENT
3. Edit the section name to Section 1

Likewise, two sections can have the same team name by doing the same method as above.

**Expected behaviour**
This should not be allowed

**Actual behaviour**
Editing is successful.

**Additional info**
Now that Section 1 has 101 students, if we try the same method to a new student, an error will be given.
image

This is caused by the following line in `UpdateStudentAction.java`
https://github.com/TEAMMATES/teammates/blob/4346f6f3d5ef8bf9570ab088a9835709f13ca99f/src/main/java/teammates/ui/webapi/action/UpdateStudentAction.java#L59
The student passed as a parameter to the method should be the **UPDATED** student attributes and not the original student attributes.

### PR DESCRIPTION
**Pull request by erjillsison**
Fixes #9752

**Outline of solution**
Previously `logic.validateSectionsAndTeams()` was incorrectly used. The student list parameter passed to that method should be the updated one and not the original values.

### CODE REVIEW

File: src/main/java/teammates/ui/webapi/action/UpdateStudentAction.java [hunk 1]
```diff
@@ -53,11 +53,17 @@ public ActionResult execute() {
...
+                .withSectionName(updateRequest.getSection())
+                .withTeamName(updateRequest.getTeam())
+                .withComment(updateRequest.getComments())
+                .build();
+
         boolean emailSent = false;
 
         try {
-            logic.validateSectionsAndTeams(Arrays.asList(student), student.course);
-            logic.validateTeams(Arrays.asList(student), student.course);
```
Comment (xpdavid) [inline src/main/java/teammates/ui/webapi/action/UpdateStudentAction.java]: I think you can remove the method?

File: src/test/java/teammates/test/cases/webapi/UpdateStudentActionTest.java [hunk 2]
```diff
@@ -177,6 +177,29 @@ public void testExecute() throws Exception {
                 invalidParamsOutput.getMessage());
     }
 
+    @Test
+    public void testExecute_withTeamNameAlreadyExistsInAnotherSection_shouldFail() {
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/webapi/UpdateStudentActionTest.java]: You should add a test that exactly test the scenario? `Should not be able to exceed section limit by editing section name of an existing student`

File: src/main/java/teammates/ui/webapi/action/UpdateStudentAction.java (hunk 1 above)
Comment (erjillsison) [reply src/main/java/teammates/ui/webapi/action/UpdateStudentAction.java]: Yup. Removed

File: src/test/java/teammates/test/cases/webapi/UpdateStudentActionTest.java (hunk 2 above)
Comment (erjillsison) [reply src/test/java/teammates/test/cases/webapi/UpdateStudentActionTest.java]: Updated with new test  case

**Comment by xpdavid** [comment]
@erjillsison Can you fix the conflict?

**Comment by erjillsison** [comment]
@xpdavid Fixed.

**Comment by erjillsison** [comment]
@xpdavid Done. I had to update a test case as it had a bug that was undetected before this PR.

**Pull request merged**